### PR TITLE
Added a timer to the KG creation pipeline

### PR DESCRIPTION
**Pull request by alexthomas93**
# Description

Adds timers to KG creation pipeline components as well as the KG pipeline as a whole.

## Type of Change
- [X] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [ ] Unit tests
- [ ] E2E tests
- [X] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [X] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: src/neo4j_genai/experimental/pipeline/pipeline.py [hunk 1]
```diff
@@ -130,6 +133,8 @@ async def execute(self, **kwargs: Any) -> RunResult | None:
             status=self.status,
             result=component_result,
         )
+        end_time = default_timer()
+        logger.info(f"Component {self.name} finished in {end_time - start_time}s")
```
Comment (willtai) [inline src/neo4j_genai/experimental/pipeline/pipeline.py:137]: Do we want this at the `logger.debug` level instead? I'm not sure if it is necessary at the `logger.info` level

Comment (alexthomas93) [reply src/neo4j_genai/experimental/pipeline/pipeline.py:137]: My understanding is that debug to give developers more fine-grained/detailed info e.g. the every step in making a request to an API and info is higher level and broader e.g. when each API request is made. So info seemed like an appropriate level for timing. In addition you get swamped with logs from all of the dependencies (huggingface, requests, etc.) at debug level so it's very hard to see these messages.

Comment (willtai) [reply src/neo4j_genai/experimental/pipeline/pipeline.py:137]: I think this granularity of time profiling of different components is not necessary at the info level i.e. if everything is working fine. It's useful if developers want to profile their code and understand the performance o feach different components imo

File: src/neo4j_genai/experimental/pipeline/pipeline.py [hunk 2]
```diff
@@ -467,8 +471,12 @@ def validate_inputs_config(self, data: dict[str, Any]) -> None:
...
 
     async def run(self, data: dict[str, Any]) -> dict[str, Any]:
+        logging.info("Starting pipeline")
+        start_time = default_timer()
         self.validate_inputs_config(data)
         self.reinitialize()
         orchestrator = Orchestrator(self)
         await orchestrator.run(data)
+        end_time = default_timer()
+        logging.info(f"Pipeline finished in {end_time - start_time}s")
```
Comment (willtai) [inline src/neo4j_genai/experimental/pipeline/pipeline.py:481]: Shall we change this to debug too?

**Pull request merged**